pkg/controller/certificate-shim/sync.go

```diff
@@ -105,26 +117,33 @@ func (c *controller) Sync(ctx context.Context, ing *networkingv1beta1.Ingress) e
 		if err != nil {
 			return err
 		}
-		c.recorder.Eventf(ing, corev1.EventTypeNormal, reasonDeleteCertificate, "Successfully deleted unrequired Certificate %q", crt.Name)
+		c.recorder.Eventf(obj, corev1.EventTypeNormal, reasonDeleteCertificate, "Successfully deleted unrequired Certificate %q", crt.Name)
 	}
 
 	return nil
 }
 
-func (c *controller) validateIngress(ing *networkingv1beta1.Ingress) []error {
-	// check for duplicate values of networkingv1beta1.IngressTLS.SecretName
-	var errs []error
-	namedSecrets := make(map[string]int)
-	for _, tls := range ing.Spec.TLS {
-		namedSecrets[tls.SecretName]++
-	}
-	// not doing this in the previous for-loop to avoid erroring more than once for the same SecretName
-	for name, n := range namedSecrets {
-		if n > 1 {
-			errs = append(errs, fmt.Errorf("Duplicate TLS entry for secretName %q", name))
+func (c *controller) validateObject(obj metav1.Object) []error {
+	switch o := obj.(type) {
+	case *networkingv1beta1.Ingress:
+		// check for duplicate values of networkingv1beta1.IngressTLS.SecretName
```

Comment:
question: why do we have to bail out when two TLS blocks contain the same secretName? I can't figure that out. This is an old code snippet but I wonder... 😅

<!-- 

what -> why

```suggestion
    // 
    // Having multiple TLS entries on an Ingress is fine, but we can't process
    // an Ingress that has two TLS entries with the same secretName.
```

-->